### ISSUE TITLE
Canceling jobs under Windows

### ISSUE THREAD

**Issue by brettbode** [brettbode/GamessQ#2]
Hi.

In XP I had a problem that there were some "ghost" processes of gamess that didn't use CPU but used a lot of page file. In Win 7 the problem is that even the CPU eating processes aren't killed when the job is cancelled.
- Antti

Brett Bode wrote:
I haven't tested Win 7, but I thought you indicated the problem originally occurred for you in XP. If this is a Win 7 specific problem it may take longer to resolve as I'll have to locate a Win 7 system I can test and debug on. It also probably means that Microsoft changed something in how processes (or threads) are managed.

Brett
On Dec 21, 2009, at 12:35 PM,  wrote:

Hi.

I did some testing and the program works fine except that in Windows 7 it still doesn't kill the processes when the job is cancelled. Have you tested it in Win 7?
- Antti

Brett Bode wrote:

Antti,
   I have posted a new version of GamessQ that we have tested locally and do not see any issues with rogue processes. Can you grab it and see if it works on your system? It isn't yet rolled into the GAMESS binary downloads so you'll have to grab it by itself at the moment.

http://www.msg.chem.iastate.edu/GAMESS/GAMESSQ/

Thanks,
Brett
On Dec 7, 2009, at 12:34 AM, wrote:

Hi.

I'm using version "12 JAN 2009 (R3)". However, I'm using a modified ddikick.exe because the official one wasn't working properly (see the comments in 
http://molecularmodelingbasics.blogspot.com/2009/07/mysterious-gamessq-program.html
 ) .

I did some testing and it seems that the problem doesn't depend on whether the job was just started or whether it had been running a while.

The latest WinGAMESS isn't fully compatible with Windows 7. At first I wasn't able to run jobs at all because Win7 isn't able to execute "runscript.csh" properly. I found the problem to be in this line "if ((-x $GMSPATH/gamess.$VERNO.exe) && (-x $GMSPATH/ddikick.exe))". Win7 doesn't understand it and just goes to else-clause and exits.
## \- Antti

Subject: Re: [FWD] Re: Cancelling jobs in windows version of GAMESSQ

Hi Antti,
 Are you using the latest version of WinGAMESS? I don't know why that would make a difference, but so far I haven't reproduced the problem in some short tests on my system. Does this happen if you cancel the job shortly after starting it, or only after the job has been running a while?

Thanks,
Brett
On Nov 30, 2009, at 4:33 AM, wrote:

Dear Dr. Bode,

The problem that I described to Jan was basically that the gamess processes aren't killed properly. What I understand is that GamessQ starts 2 gamess processes per processor: one that is using a lot CPU and one that isn't. Is that correct? What I noticed is that the second process isn't always killed when the job is cancelled and that process sometimes uses a lot of page file. All that was in WinXP (with the latest updates) with GamessQ 1.1. I was mostly running simple RUNTYP=OPTIMIZE jobs.

I have since switched to Windows 7 Professional and now I have another problem. Cancelling a running job in GamessQ doesn't work. It says "Canceled" as the status but the job is still running. I can even use "View Logs" for those cancelled jobs.

Regarding the reorder idea that Jan mentions, I think it would be useful to be able to move jobs up and down in the queue. Also, it would be nice to have more columns, for example "Energy" and "Maximum gradient". That would make it easier to monitor jobs.

I'd be happy to do more testing if you have any suggestions.

Best regards,

Antti
